File: DelegateSample/DelegateSample/ViewController.m
//
//  ViewController.m
//  DelegateSample
//
//  Created by mac mini on 9/30/14.
//  Copyright (c) 2014 ParadigmCreatives. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
	// Do any additional setup after loading the view, typically from a nib.
    
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}
-(BOOL)textFieldShouldReturn:(UITextField *)textField
{
    [_nameTF resignFirstResponder];
    [_passTF resignFirstResponder];
    return YES;
}
-(void)delegateFunction:(NSString *)textfield;
{
    self.nameTF.text=textfield;
}
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    if([[segue identifier]isEqualToString:@"pushToSecond"])
    {
        SecondViewController *second=[segue destinationViewController];
        second.delegateObject=self;
    }
}
@end

File: DelegateSample copy/DelegateSample/ViewController.swift
//
//  ViewController.swift
//  DelegateSample
//
//  Created by SivaSankar on 24/10/14.
//  Copyright (c) 2014 Paradim Creatives. All rights reserved.
//

import UIKit

class ViewController: UIViewController ,UpdateViewControllerDelegate{

    @IBOutlet var nameTF: UITextField!
    @IBOutlet var passwdTF: UITextField!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
}

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        if segue.identifier == "push"
        {
            let viewController = segue.destinationViewController as UpdateViewController
            viewController.delegate=self
        }
    }

    @IBAction func nextClicked(sender: AnyObject) {
    }

    func function(controller:UpdateViewController,textString:String){
    
        nameTF.text=textString
         controller.navigationController?.popToRootViewControllerAnimated(true)
    }
}

